test/BreakInvariantBounty.test.js

```diff
@@ -17,82 +17,95 @@ const sendReward = async (from, to, value) => ethSendTransaction({
-
-        const balance = await ethGetBalance(this.bounty.address);
-        balance.should.be.bignumber.equal(reward);
-      });
+contract('BreakInvariantBounty', function ([_, owner, researcher, anyone, nonTarget]) {
+  beforeEach(async function () {
+    this.bounty = await BreakInvariantBountyMock.new({ from: owner });
+  });
 
-      it('cannot claim reward', async function () {
-        await assertRevert(
-          this.bounty.claim(this.targetAddress, { from: researcher }),
-        );
-      });
-    });
+  it('can set reward', async function () {
+    await sendReward(owner, this.bounty.address, reward);
+    (await ethGetBalance(this.bounty.address)).should.be.bignumber.equal(reward);
   });
 
-  context('against broken contract', function () {
+  context('with reward', function () {
     beforeEach(async function () {
-      this.bounty = await InsecureInvariantTargetBounty.new();
-
-      const result = await this.bounty.createTarget({ from: researcher });
-      const event = expectEvent.inLogs(result.logs, 'TargetCreated');
-
-      this.targetAddress = event.args.createdAddress;
       await sendReward(owner, this.bounty.address, reward);
     });
 
-    it('can claim reward', async function () {
-      await this.bounty.claim(this.targetAddress, { from: researcher });
-      const claim = await this.bounty.claimed();
-
-      claim.should.equal(true);
-
-      const researcherPrevBalance = await ethGetBalance(researcher);
+    describe('destroy', function () {
+      it('returns all balance to the owner', async function () {
+        const ownerPreBalance = await ethGetBalance(owner);
+        await this.bounty.destroy({ from: owner, gasPrice: 0 });
+        const ownerPostBalance = await ethGetBalance(owner);
 
-      await this.bounty.withdrawPayments(researcher, { gasPrice: 0 });
-      const updatedBalance = await ethGetBalance(this.bounty.address);
-      updatedBalance.should.be.bignumber.equal(0);
+        (await ethGetBalance(this.bounty.address)).should.be.bignumber.equal(0);
+        ownerPostBalance.sub(ownerPreBalance).should.be.bignumber.equal(reward);
+      });
 
-      const researcherCurrBalance = await ethGetBalance(researcher);
-      researcherCurrBalance.sub(researcherPrevBalance).should.be.bignumber.equal(reward);
+      it('reverts when called by anyone', async function () {
+        await assertRevert(this.bounty.destroy({ from: anyone }));
+      });
     });
 
-    it('cannot claim reward from non-target', async function () {
-      await assertRevert(
-        this.bounty.claim(nonTarget, { from: researcher })
-      );
-    });
+    describe('claim', function () {
+      it('is initially unclaimed', async function () {
+        (await this.bounty.claimed()).should.equal(false);
+      });
+
+      it('can create claimable target', async function () {
+        const { logs } = await this.bounty.createTarget({ from: researcher });
+        expectEvent.inLogs(logs, 'TargetCreated');
```

Comment:
for the future, I think the API here is incomplete. We are creating a target but we have no way to get it afterwards. So the only way to check it is through the logs, which I think is incorrect.

Reply:
Thing is, afaik all non-`view` function calls return a transaction receipt when called from `truffle-contracts`, so while the function does return that (and, say, a contract could get that value), we can only get it in the events. Regarding having some additional means of retrieving it, we'd have to introduce some additional data structure to keep track of these, since as of now it is not possible. We may want to change this anyway due to #1333.

Reply:
yeah, I mean to add those additional structures. I'm proposing to add a little complexity in order to be able to retrieve everything we set at any time. Just a design principle we can take into account when building the API, nothing to do right now.